src/include/timestamp.i

```diff
@@ -0,0 +1,196 @@
+/*-
+ * Copyright (c) 2014-2020 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * __wt_time_window_init --
+ *     Initialize the fields in a time window to their defaults.
+ */
+static inline void
+__wt_time_window_init(WT_TIME_WINDOW *tw)
+{
+    tw->start_durable_ts = WT_TS_NONE;
+    tw->start_ts = WT_TS_NONE;
+    tw->start_txn = WT_TXN_NONE;
+    tw->stop_durable_ts = WT_TS_NONE;
+    tw->stop_ts = WT_TS_MAX;
+    tw->stop_txn = WT_TXN_MAX;
+    tw->prepare = false;
+}
+
+/*
+ * __wt_time_window_init_max --
+ *     Initialize the fields in a time window to values that force an override.
+ */
+static inline void
+__wt_time_window_init_max(WT_TIME_WINDOW *tw)
+{
+    tw->start_durable_ts = WT_TS_MAX;
+    tw->start_ts = WT_TS_MAX;
+    tw->start_txn = WT_TXN_MAX;
+    tw->stop_durable_ts = WT_TS_MAX;
+    tw->stop_ts = WT_TS_NONE;
+    tw->stop_txn = WT_TXN_NONE;
+    tw->prepare = false;
+}
+
+/*
+ * __wt_time_window_copy --
+ *     Copy the values from one time window structure to another.
+ */
+static inline void
+__wt_time_window_copy(WT_TIME_WINDOW *dest, WT_TIME_WINDOW *source)
+{
+    /* Don't use memory copy - this is in performance critical paths */
+    dest->start_durable_ts = source->start_durable_ts;
+    dest->stop_durable_ts = source->stop_durable_ts;
+    dest->start_ts = source->start_ts;
+    dest->start_txn = source->start_txn;
+    dest->stop_ts = source->stop_ts;
+    dest->stop_txn = source->stop_txn;
+    dest->prepare = source->prepare;
```

Comment:
I'd use structure assignment here, I believe current compilers do the right thing.